server/src/main/java/org/candlepin/dto/api/v1/PoolDTO.java

```diff
@@ -0,0 +1,1846 @@
+        if (calculatedAttributes != null) {
+            if (this.calculatedAttributes == null) {
+                this.calculatedAttributes = new HashMap<String, String>();
+            }
+            else {
+                this.calculatedAttributes.clear();
+            }
+            this.calculatedAttributes.putAll(calculatedAttributes);
+        }
+        else {
+            this.calculatedAttributes = null;
+        }
+        return this;
+    }
+
+    /**
+     * Returns the upstream pool id.
+     *
+     * @return the upstream pool id.
+     */
+    public String getUpstreamPoolId() {
+        return upstreamPoolId;
+    }
+
+    /**
+     * Sets the upstream pool id.
+     *
+     * @param upstreamPoolId set the upstream pool id.
+     *
+     * @return a reference to this PoolDTO object.
+     */
+    public PoolDTO setUpstreamPoolId(String upstreamPoolId) {
+        this.upstreamPoolId = upstreamPoolId;
+        return this;
+    }
+
+    /**
+     * Returns the upstream entitlement id.
+     *
+     * @return the upstream entitlement id.
+     */
+    public String getUpstreamEntitlementId() {
+        return upstreamEntitlementId;
+    }
+
+    /**
+     * Sets the upstream entitlement id.
+     *
+     * @param upstreamEntitlementId set the upstream entitlement id.
+     *
+     * @return a reference to this PoolDTO object.
+     */
+    public PoolDTO setUpstreamEntitlementId(String upstreamEntitlementId) {
+        this.upstreamEntitlementId = upstreamEntitlementId;
+        return this;
+    }
+
+    /**
+     * Returns the upstream consumer id.
+     *
+     * @return the upstream consumer id.
+     */
+    public String getUpstreamConsumerId() {
+        return upstreamConsumerId;
+    }
+
+    /**
+     * Sets the upstream consumer id.
+     *
+     * @param upstreamConsumerId set the upstream consumer id.
+     *
+     * @return a reference to this PoolDTO object.
+     */
+    public PoolDTO setUpstreamConsumerId(String upstreamConsumerId) {
+        this.upstreamConsumerId = upstreamConsumerId;
+        return this;
+    }
+
+    /**
+     * The Marketing/Operations product name for the
+     * <code>productId</code>.
+     *
+     * @return the productName
+     */
+    @JsonProperty
+    public String getProductName() {
+        return productName;
+    }
+
+    /**
+     * Set the Marketing/Operations product name for the
+     * <code>productId</code>.
+     *
+     * @param productName set the productName
+     *
+     * @return a reference to this PoolDTO object.
+     */
+    @JsonIgnore
+    public PoolDTO setProductName(String productName) {
+        this.productName = productName;
+        return this;
+    }
+
+    /**
+     * Return the "top level" product this pool is for.
+     * Note that pools can also provide access to other products.
+     * See getProvidedProducts().
+     *
+     * @return Top level product ID.
+     */
+    public String getProductId() {
```

Comment:
We definitely don't need both product ID+name and the product object. We should pick one or the other.